src/server/scriptInfo.ts

```diff
@@ -68,37 +77,51 @@ namespace ts.server {
             this.ownFileText = false;
             this.text = undefined;
             this.lineMap = undefined;
+            this.fileSize = undefined;
         }
 
         /**
          * Set the contents as newText
          * returns true if text changed
          */
-        public reload(newText: string) {
+        public reload(newText: string): boolean {
             Debug.assert(newText !== undefined);
 
             // Reload always has fresh content
             this.pendingReloadFromDisk = false;
 
+            // We only need both this.text and this.fileSize if this.text
+            // is artificially empty because it was too large.
+            // We assume that an empty string passed to reload is not
+            // for a file that was too large to store.
+            this.fileSize = undefined;

```

Comment:
I don't think you should set this to undefined here. This should happen at reload callsites because that's where the information of actual file size lies.

Reply:
I don't understand your suggestion.

Reply:
I moved it into `useText`.